src/internal/memory/device_resources.hpp

```diff
@@ -0,0 +1,51 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2021-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "internal/resources/partition_resources_base.hpp"
+#include "internal/runnable/resources.hpp"
+
+#include "srf/memory/buffer.hpp"
+#include "srf/memory/resources/memory_resource.hpp"
+
+#include <cstddef>
+#include <memory>
+#include <optional>
+
+namespace srf::internal::ucx {
+class Resources;
+}
+
+namespace srf::internal::memory {
+
```

Comment:
Could use at least a top level comment block describing its intended function and purpose.